### PR TITLE
feat(desktop): expand DesktopAPI bridge with browse_file, save_file, open_path

### PR DESCRIPTION
**Pull request by curdriceaurora**
## Summary

- Adds three new `DesktopAPI` methods (`browse_file`, `save_file`, `open_path`) giving the web UI native file-picker and file-reveal capabilities when running inside pywebview
- Ships `desktop_api.js` with browser-safe JS wrappers and desktop mode detection; integrates into templates with `[data-desktop-only]` CSS gating
- 32 unit tests (100% diff coverage on new Python lines); docs updated with full bridge API reference

### Python bridge (`src/file_organizer/desktop/app.py`)
- `browse_file(file_types)` — native file-picker via `OPEN_DIALOG`
- `save_file(suggested_name, file_types)` — Save-As via `SAVE_DIALOG`; strips `/` and `\` from `suggested_name` (F4 path-separator injection guard)
- `open_path(path)` — reveals path in OS file manager; platform dispatch (macOS: `open -R`, Windows: `explorer /select,…`, Linux: `xdg-open parent`); always `shell=False` (F4); resolves path before forwarding
- Module-level `import subprocess`, `import sys` for clean `patch()` test patching

### JS utility (`src/file_organizer/web/static/js/desktop_api.js`)
- `window.desktopBrowseFile(inputId, fileTypes)` — calls pywebview API or falls back to hidden `<input type="file">`
- `window.desktopSaveFile(suggestedName, fileTypes)` — calls pywebview API or returns `""`
- `window.desktopOpenPath(path)` — calls pywebview API or silent no-op
- Sets `document.body.dataset.desktopApp="1"` on load when `window.pywebview` is present

### Template integration
- `base.html`: loads `desktop_api.js` before `app.js`
- `files/_results.html`: "Reveal in Files" context-menu item (`data-desktop-only`)
- `settings/index.html`: desktop Browse button + read-only path input (`data-desktop-only`)
- `styles.css`: `[data-desktop-only]` hidden in browser mode, shown via `body[data-desktop-app="1"]`

### Unit tests (32 tests, all `@pytest.mark.unit @pytest.mark.ci`)
- `tests/desktop/test_desktop_api_browse_file.py` — 10 tests
- `tests/desktop/test_desktop_api_save_file.py` — 10 tests
- `tests/desktop/test_desktop_api_open_path.py` — 12 tests

### Docs
- `docs/desktop-app.md`: new "Python–JavaScript Bridge API" section with methods table, `file_types` format, and security guarantees

## Test plan

- [x] All 32 new unit tests pass (`pytest tests/desktop/ -v`)
- [x] Pre-commit validation passes (ruff, coverage, diff-cover 100%, docstring gate)
- [x] No absolute paths in test files (G1 hook)
- [x] `open_path` shell=False verified by 3 platform-specific tests
- [x] `save_file` path-separator stripping verified by 2 security tests

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Native file/folder browse, native "save as", and reveal/open-in-file-manager actions in the desktop app
  * Desktop-only UI elements (toggled in desktop mode), a context-menu "Reveal in Files" action, and a desktop settings-file picker

* **Documentation**
  * New guide and API reference for the desktop JS–Python bridge, dialog behaviors, file-type formats, and security guarantees

* **Tests**
  * Unit tests covering browse, save, and open-path desktop behaviors and edge cases

### CODE REVIEW

File: src/file_organizer/web/templates/files/_results.html [hunk 1]
```diff
@@ -197,6 +197,12 @@
                         <button type="button" data-context-action="rename" disabled title="Rename is not implemented yet">Rename</button>
                         <button type="button" data-context-action="move" disabled title="Move is not implemented yet">Move</button>
                         <button type="button" data-context-action="delete" disabled title="Delete is not implemented yet">Delete</button>
+                        <button
+                            type="button"
+                            data-context-action="reveal"
+                            data-desktop-only
+                            onclick="window.desktopOpenPath('{{ entry.path }}')"
+                        >Reveal in Files</button>
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply src/file_organizer/web/templates/files/_results.html]: Fixed: Replaced inline JS interpolation with a data-open-path attribute on the button; onclick now reads this.dataset.openPath so the path is never injected into a JS string literal.
